Emrald-UI/src/contexts/StateContext.tsx

```diff
@@ -174,6 +228,7 @@ const StateContextProvider: React.FC<EmraldContextWrapperProps> = ({ children })
 
   const clearStateList = () => {
     setStates([]);
+    appData.value.StateList = [];
```

Comment:
I'm not sure if this is how I was doing this in any of the other context files but we should avoid updating the appData.value items directly. Ideally it should go through the updateAppData function so that the session storage is updated also. With this approach those changes won't be applied to the session storage.
```suggestion
    updateAppData({...appData.value, StateList: []);
```